hyasinsight_connector.py

```diff
@@ -0,0 +1,1262 @@
+        # the action for this param
+        endpoint = PASSIVEHASH
+
+        # Access action parameters passed in the 'param' dictionary
+
+        # Required values can be accessed directly
+        self.save_progress("Checking the Indicator value")
+        self.handle_all_actions(param, endpoint, indictor_type)
+
+    def _handle_lookup_ssl_certificate_ip(self, param):
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the
+        # platform
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier()))
+        indictor_type = IP
+        # Add an action result object to self (BaseConnector) to represent
+        # the action for this param
+        endpoint = SSL
+
+        # Access action parameters passed in the 'param' dictionary
+
+        # Required values can be accessed directly
+        self.save_progress("Checking the Indicator value")
+        self.handle_all_actions(param, endpoint, indictor_type)
+
+    def _handle_lookup_passivedns_domain(self, param):
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the
+        # platform
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier()))
+        indictor_type = DOMAIN
+        # Add an action result object to self (BaseConnector) to represent
+        # the action for this param
+        endpoint = PASSIVEDNS
+
+        # Access action parameters passed in the 'param' dictionary
+
+        # Required values can be accessed directly
+        self.save_progress("Checking the Indicator value")
+        self.handle_all_actions(param, endpoint, indictor_type)
+
+    def _handle_lookup_current_whois_domain(self, param):
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the
+        # platform
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier()))
+        indictor_type = DOMAIN
+        # Add an action result object to self (BaseConnector) to represent
+        # the action for this param
+        endpoint = WHOIS
+
+        # Access action parameters passed in the 'param' dictionary
+
+        # Required values can be accessed directly
+        self.save_progress("Checking the Indicator value")
+        self.handle_all_actions(param, endpoint, indictor_type)
+
+    def _handle_lookup_passivedns_ip(self, param):
+        # Implement the handler here
+        # use self.save_progress(...) to send progress messages back to the
+        # platform
+        self.save_progress(
+            "In action handler for: {0}".format(self.get_action_identifier()))
+        indictor_type = IPV4
+        # Add an action result object to self (BaseConnector) to represent
+        # the action for this param
+        endpoint = PASSIVEDNS
```

Comment:
Seems like the functions between line 311 and here are all the same style. Maybe a single function with a mapping would work out. This is not a requirement, but a suggestion.